src/Autofac/Core/Activators/Reflection/ConstructorBinder.cs

```diff
@@ -36,7 +36,7 @@ public ConstructorBinder(ConstructorInfo constructorInfo)
         if (_illegalParameter is null)
         {
             // Build the invoker.
-            _factory = FactoryCache.GetOrAdd(constructorInfo, FactoryBuilder);
+            _factory = FactoryCache.GetOrAdd(Constructor, FactoryBuilder);
```

Comment:
Should this use the `GetConstructorInvoker` new method?

```suggestion
            _factory = GetConstructorInvoker();
```